spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/config/JdbcRepositoryConfigExtension.java

```diff
@@ -87,103 +87,17 @@ public void registerBeansForRoot(BeanDefinitionRegistry registry, RepositoryConf
 	 */
 	@Override
 	public void postProcess(BeanDefinitionBuilder builder, RepositoryConfigurationSource source) {
 
-		resolveReference(builder, source, "jdbcOperationsRef", "jdbcOperations", NamedParameterJdbcOperations.class, true);
-		resolveReference(builder, source, "dataAccessStrategyRef", "dataAccessStrategy", DataAccessStrategy.class, false);
-	}
-
-	private void resolveReference(BeanDefinitionBuilder builder, RepositoryConfigurationSource source,
-			String attributeName, String propertyName, Class<?> classRef, boolean required) {
-
-		Optional<String> beanNameRef = source.getAttribute(attributeName).filter(StringUtils::hasText);
-
-		String beanName = beanNameRef.orElseGet(() -> determineMatchingBeanName(propertyName, classRef, required));
-
-		if (beanName != null) {
-			builder.addPropertyReference(propertyName, beanName);
-		} else {
-			Assert.isTrue(!required,
-					"The beanName must not be null when requested as 'required'. Please report this as a bug.");
-		}
-
-	}
-
-	@Nullable
-	private String determineMatchingBeanName(String propertyName, Class<?> classRef, boolean required) {
-
-		if (this.beanFactory == null) {
-			return nullOrThrowException(required,
-					() -> new NoSuchBeanDefinitionException(classRef, "No BeanFactory available."));
-		}
-
-		List<String> beanNames = Arrays.asList(beanFactory.getBeanNamesForType(classRef));
-
-		if (beanNames.isEmpty()) {
-			return nullOrThrowException(required,
-					() -> new NoSuchBeanDefinitionException(classRef, String.format("No bean of type %s available", classRef)));
-		}
-
-		if (beanNames.size() == 1) {
-			return beanNames.get(0);
-		}
-
-		if (!(beanFactory instanceof ConfigurableListableBeanFactory)) {
-
-			return nullOrThrowException(required,
-					() -> new NoSuchBeanDefinitionException(String.format(
-							"BeanFactory does not implement ConfigurableListableBeanFactory when trying to find bean of type %s.",
-							classRef)));
-		}
-
-		List<String> primaryBeanNames = getPrimaryBeanDefinitions(beanNames, (ConfigurableListableBeanFactory) beanFactory);
-
-		if (primaryBeanNames.size() == 1) {
-			return primaryBeanNames.get(0);
-		}
-
-		if (primaryBeanNames.size() > 1) {
-			throw new NoUniqueBeanDefinitionException(classRef, primaryBeanNames.size(),
-					"more than one 'primary' bean found among candidates: " + primaryBeanNames);
-		}
-
-		for (String beanName : beanNames) {
-
-			if (propertyName.equals(beanName)
-					|| ObjectUtils.containsElement(beanFactory.getAliases(beanName), propertyName)) {
-				return beanName;
-			}
-		}
-
-		return nullOrThrowException(required,
-				() -> new NoSuchBeanDefinitionException(String.format("No bean of name %s found.", propertyName)));
+		builder.addPropertyValue("jdbcOperationsRef", source.getAttribute("jdbcOperationsRef").orElse(null));
```

Comment:
Let's keep the notion of conditional configuration here. If a bean name is declared, then we should set and configure a property value. Otherwise, if a default applies, then we don't configure that property and let the factory bean do its default.

Reply:
The benefit is that we've got a split of responsibilities. The config extension is responsible for configuring a property if the annotation indicates to do so. If the annotation indicates defaulting, then nothing is configured and the issue is handled in a downstream component.